File: lib/data/core/telleo_logger.dart
import 'package:injectable/injectable.dart';
import 'package:logger/logger.dart';

import '../../domain/core/services/logger.dart';

@LazySingleton(as: ILogger)
class TelleoLogger implements ILogger {
  final Logger logger;
  TelleoLogger({
    required this.logger,
  });

  @override
  void logError(error) {
    logger.e(error.toString());
  }

  @override
  void logInfo(String info) {
    logger.i(info);
  }

  @override
  void logWarning(String warning) {
    logger.w(warning);
  }
}

File: backend/src/core/data/logging/telleo_logger.ts
import winston from "winston";
import Logger from "../../domain/logging/logger";

export default class TelleoLogger implements Logger {
  private log = winston.createLogger({
    format: winston.format.json(),
    transports: [
      new winston.transports.Console({
        format: winston.format.simple(),
      }),
    ],
  });

  warning(warning: any): void {
    this.log.warn(warning);
  }

  info(info: any): void {
    this.log.info(info);
  }

  error(error: any): void {
    this.log.error(error);
  }
}
